File: Code Examples/Security_OTP_Example/src/main/java/io/hashimati/security/repository/UserRepository.java
package io.hashimati.security.repository;


import io.hashimati.security.domains.LoginStatus;
import io.hashimati.security.domains.User;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;

import java.time.Instant;
import java.util.List;

@JdbcRepository(dialect = Dialect.H2)
public interface UserRepository extends CrudRepository<User, Long> {
    public User findByUsername(String username);
    public List<User> findAll();
    public boolean existsByUsername(String username);
    public Long updateByUsername(String username, LoginStatus lastLoginStatus, Instant lastTimeTryToLogin);
    public Long updateByUsername(String username, String oneTimePassword, Instant oneTimePasswordExpiry);
}

File: microstartercli/src/main/resources/micronaut/security/jwt/kotlin/JDBC/repository/UserRepository.kt
package ${securityPackage}.repository

import ${securityPackage}.domains.LoginStatus
import ${securityPackage}.domains.User
<% if(jdbc) out.print 'import io.micronaut.data.jdbc.annotation.JdbcRepository'%><% if(mongo) out.print 'import io.micronaut.data.mongodb.annotation.MongoRepository'%>
<% if(jdbc) out.print 'import io.micronaut.data.model.query.builder.sql.Dialect'%>
import io.micronaut.data.repository.CrudRepository
import java.time.Instant

<% if(jdbc) out.print '@JdbcRepository(dialect = Dialect.'+dialect+')'%><% if(mongo) out.print '@MongoRepository'%>
interface UserRepository : CrudRepository<User, <% if(jdbc) out.print 'Long'%><% if(mongo) out.print 'String'%>> {
    fun findByUsername(username: String?): User?
    override fun findAll(): List<User?>
    fun existsByUsername(username: String?): Boolean
    fun updateByUsername(username: String?, lastLoginStatus: LoginStatus?, lastTimeTryToLogin: Instant?): Long?
}